src/three-domain-secure/component.jsx

```diff
@@ -1,30 +1,52 @@
 /* @flow */
-import { getLogger, getSDKToken } from "@paypal/sdk-client/src";
+import { type LoggerType } from "@krakenjs/beaver-logger/src";
 import { FPTI_KEY } from "@paypal/sdk-constants/src";
 
 import { ValidationError } from "../lib";
 
-export const getThreeDomainSecure = (): Function => {
-  const sdkToken = getSDKToken();
-  const ThreeDomainSecureAuth = () => {
-    if (sdkToken) {
-      // eslint-disable-next-line no-console
-      console.log("Three Domain Secure Called");
-      // Make a Zoid component and introduce methods here
-      // onSuccess
-      // onCancel
-      // onClose
-      getLogger()
-        .info("three domain secure v2 invoked")
-        .track({
-          [FPTI_KEY.TRANSITION]: "three_DS_auth_v2",
-        });
-    } else {
-      throw new ValidationError(
-        `script data attribute sdk-client-token is required but was not passed`
-      );
-    }
-  };
-
-  return ThreeDomainSecureAuth;
+type SdkConfig = {|
+  sdkToken: ?string,
+|};
+
+const parseSdkConfig = ({ sdkConfig, logger }): SdkConfig => {
+  if (!sdkConfig.sdkToken) {
+    throw new ValidationError(
+      `script data attribute sdk-client-token is required but was not passed`
+    );
+  }
+
+  logger.info("three domain secure v2 invoked").track({
+    [FPTI_KEY.TRANSITION]: "three_DS_auth_v2",
+  });
+
+  return sdkConfig;
 };
+export interface ThreeDomainSecureComponentInterface {
+  isEligible(): void;
```

Comment:
nit: do we want this interface to go ahead and reflect the future state by returning a boolean? Then could update line 43 to just always return `false` for now. The test for it would then be closer to the design too (and not test for a mocked `console.log`.

Reply:
This is a good call out. I modified the interface for both `isEligible()` and `show()` to better reflect the future state.